src/main/java/com/band/api/resolvers/Mutation.java

```diff
@@ -26,24 +33,57 @@ public Mutation(UserRepository userRepository){
      * @param name
      * @return the created User
      */
-    public User createUser(String email, String username, String password, String name){
+    public User createUser(String email, String username, String password, String name) throws CustomException{
+        User user = null;
+        CustomException ex = null;
 
-        BCryptPasswordEncoder bc = new BCryptPasswordEncoder();
-        String passwordHash = bc.encode(password);
-        User user = User.builder()
-                    .email(email)
+        try {
+            verifyUsernameEmail(username,email);
+            String passwordHash = encoder.encode(password);
+            user = User.builder()
+                    .emailSearch(email.toLowerCase())
+                    .emailDisplay(email)
                     .username(username)
                     .passwordHash(passwordHash)
                     .name(name)
                     .build();
-        try {
             userRepository.save(user);
         }
+        catch (DataAccessResourceFailureException e){
+            //Database Unavailable
+            ex = new CustomException("Database Unavailable");
+        }
+        catch (CustomException e){
+            ex = e;
+        }
         catch (Exception e){
-            //TODO return some error
-            //Possible exceptions: Duplicate username or email.
-            return new User();
+            ex = new CustomException(e.getMessage());
+        }
+        if(ex != null){
+            log.debug(ex.toString());
+            throw ex;
         }
         return user;
     }
+    private void verifyUsernameEmail(String username, String email){
```

Comment:
Probably should rename to something like "validateNewUser(...)".

Reply:
Updated to verifyNewUser(...) to verify that the information entered can create a new user.